### ISSUE TITLE
Metadata parsing does not work for USGS landsat collection

### ISSUE THREAD

**Issue by Kirill888** [opendatacube/odc-stac#74]
minimal reproducer:

```python
import odc.stac
import pystac.item

item = pystac.item.Item.from_file("https://landsatlook.usgs.gov/stac-server/collections/landsat-c2l2-sr/items/LC09_L2SR_188061_20220621_20220624_02_T1_SR")
odc.stac.load([item], chunks={})
```

Fails to resolve band name/alias. 
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Input In [5], in <cell line: 5>()
      2 import pystac.item
      4 item = pystac.item.Item.from_file("https://landsatlook.usgs.gov/stac-server/collections/landsat-c2l2-sr/items/LC09_L2SR_188061_20220621_20220624_02_T1_SR")
----> 5 odc.stac.load([item], chunks={})

File /srv/conda/envs/notebook/lib/python3.8/site-packages/odc/stac/_load.py:492, in load(items, bands, groupby, resampling, dtype, chunks, pool, crs, resolution, align, geobox, bbox, lon, lat, x, y, like, geopolygon, progress, stac_cfg, patch_url, preserve_original_order, **kw)
    489 # Check we have all the bands of interest
    490 # will raise ValueError if no such band/alias
    491 collection = _collection(_parsed)
--> 492 bands_to_load = collection.resolve_bands(bands)
    493 bands = list(bands_to_load)
    495 load_cfg = _resolve_load_cfg(
    496     bands_to_load,
    497     resampling,
   (...)
    500     nodata=kw.get("nodata", None),
    501 )

File /srv/conda/envs/notebook/lib/python3.8/site-packages/odc/stac/_model.py:118, in RasterCollectionMetadata.resolve_bands(self, bands)
    112 def resolve_bands(
    113     self, bands: Optional[Union[str, Sequence[str]]] = None
    114 ) -> Dict[str, RasterBandMetadata]:
    115     """
    116     Query bands taking care of aliases.
    117     """
--> 118     return _resolve_aliases(self.bands, self.aliases, bands)

File /srv/conda/envs/notebook/lib/python3.8/site-packages/odc/stac/_model.py:407, in _resolve_aliases(src, aliases, bands)
    405     src_name = aliases.get(name, name)
    406     if src_name not in src:
--> 407         raise ValueError(f"No such band or alias: '{name}'")
    408     out[name] = src[src_name]
    410 return out

ValueError: No such band or alias: 'coastal'
```

Metadata parsing seems to assume that canonical name of a band is the same as the key in the assets table, when really `eo:band:name` should be used. In this collection, asset names match "common name" of the band not canonical name, causing error above.

Originally reported in #73.

Metadata parsing should be extended to support arbitrary band vs asset name and also to support multiple bands per asset.

**Comment by xxie12** [comment]
Hi, I post #73, is there any way I can temporarily solve this band's name problem? I am confused about what name I should use at this point. Thanks.

**Comment by Kirill888** [comment]
> Hi, I post #73, is there any way I can temporarily solve this band's name problem? I am confused about what name I should use at this point. Thanks.

version `odc-stac==0.2.4` does not have this problem, so you can try that

**Comment by Kirill888** [comment]
@xxie12 another workaround is to override aliases for bands to point to asset name rather than canonical name, this works and auto-finds CRS/resolution too. 

```python
cfg={'landsat-c2l2-sr': {'aliases': 
  {k:k for k in ['coastal', 'blue', 'green', 'red', 'nir08', 'swir16', 'swir22']}}}

... = odc.stac.load(..., bands=['red', 'green',...], stac_cfg=cfg)
```

**Comment by Kirill888** [comment]
Should be fixed by #77
